Support sharable configurations

jonathantneal opened the pull request:
A configuration file can extend the set of enabled rules from base
configurations.

1. If an `extends` option is truthy, then;
   1. Declare `opts.config` to be `echint-config-${ extends }`-like.

ahmadnassri commented:
mind updating the `README` file to reflect this behavior as well?

jonathantneal commented:
I don’t mind at all. I’ve gone ahead and updated the request.

ahmadnassri commented:
excellent :+1: 

how about some tests to better illustrate this usage? _(sorry should have asked the first time)_

jonathantneal commented:
Sure thing. A test for using a sharable configuration (based on the config test) has been added.

jonathantneal committed:
Support sharable configurations

A configuration file can extend the set of enabled rules from base
configurations.

1. If an extends option is truthy, then;
   1. Declare opts.config to be echint-config-${ extends }-like.

ahmadnassri commented:
thanks for this!
